scripts/ci_build_and_test_in_container.sh

```diff
@@ -80,11 +80,16 @@ if [[ "$*" == *--test-documentation* ]]; then
 fi
 
 # "Make" target check (builds geosx executable target only if true)
-# Use one process to prevent out-of-memory error
+# Reduce n° of processes to prevent out-of-memory error
+if [ ${CMAKE_BUILD_TYPE} == 'Debug' ]; then
+  np=2
+else
+  np=$(nproc)
+fi
```

Comment:
Do you think you could be more precise by using job pools https://cmake.org/cmake/help/latest/prop_gbl/JOB_POOLS.html

Also, is it for all `Debug` builds or only `CUDA` build (e.g.)?
Maybe you can configure the amount of cores used per build directly in the `yaml` files using the pools?

Reply:
I do not know what the best solution is about that...

Were you able to locate where/when the issue was?
Would using a job pool of `1` make sense?
Could it be possible to run the `ninja ...` in different steps, maybe triggering the compilation(s) unit(s) which is/are at stake first (like `ninja path/to/my_big_file.cpp.o path/to/my_other_big_file.cpp.o`) and then a std `ninja` with no proc restriction?